installbuilder/datafiles/Base_DSC.data

```diff
@@ -243,15 +231,31 @@ chown -R ${{RUN_AS_USER}} /opt/microsoft/omsagent/plugin
 # If this is ever changed/removed, coordinate with nxOMSAuditdPlugin author/maintainer(s)
 chown -R ${{RUN_AS_USER}} /opt/microsoft/omsconfig/Scripts
 
-# Set up the modules
+# Set up built-in resource modules for OMS DSC
 su - omsagent -c "/opt/microsoft/omsconfig/Scripts/InstallModule.py /opt/microsoft/omsconfig/module_packages/nx_1.0.zip 0"
 su - omsagent -c "/opt/microsoft/omsconfig/Scripts/InstallModule.py /opt/microsoft/omsconfig/module_packages/nxOMSPerfCounter_2.2.zip 0"
 su - omsagent -c "/opt/microsoft/omsconfig/Scripts/InstallModule.py /opt/microsoft/omsconfig/module_packages/nxOMSSyslog_2.2.zip 0"
 su - omsagent -c "/opt/microsoft/omsconfig/Scripts/InstallModule.py /opt/microsoft/omsconfig/module_packages/nxOMSKeyMgmt_1.0.zip 0"
 su - omsagent -c "/opt/microsoft/omsconfig/Scripts/InstallModule.py /opt/microsoft/omsconfig/module_packages/nxFileInventory_1.1.zip 0"
 
+# Set up a built-in resource module for DIY DSC that was removed during OMS Agent update
+# This is a temporary workaround to prevent DIY DSC from breaking after the OMS Agent is updated from version 1.4.3-225 or below
+# This section can be removed when telemetry shows that customers no longer need to update from version 1.4.3-225 or below
+
+# Check if DIY DSC install module script is present on the machine 
+if [ -f /opt/microsoft/dsc/Scripts/InstallModule.py ]; then
+
+  # Check if nx module package is available in DIY DSC
+  if [ -f /opt/microsoft/dsc/module_packages/nx_1.0.zip ]; then
+    /opt/microsoft/dsc/Scripts/InstallModule.py /opt/microsoft/dsc/module_packages/nx_1.0.zip 0
+  fi
+fi
```

Comment:
what's the purpose of this workaround ? if with new omsagent we are only installing and removing renamed binaries then when will this code executed with next release. is the intension to automatically fix broken DSC DYI for machines where upgrade between already released omsagents broke things ?

Reply:
In the case of customers that had OMS Agent installed, and they then installed DIY DSC over it, since the old package will run remove on extension upgrade, it will remove the overlapping resources (Package, User, Service, Group) from the box which will break DIY DSC again in this update. This workaround is to prevent that by re-installing the DIY DSC nx module if it the DIY DSC install script and nx module are available when we then install the new OMS Agent package.

Reply:
yes above will be needed. what about the scenario where someone will run upgrade on new agent without running remove of old one. In that case we will have left over binaries correct ?